### PR TITLE
gh/actions: Customize cilium-config

### PR DESCRIPTION
**Pull request by brb**
See commit msg.

### CODE REVIEW

**Review by jschwinger233 — APPROVED** [review]

**Review by aanm — APPROVED** [review]

File: .github/actions/cilium-config/action.yml [hunk 1]
```diff
@@ -63,7 +69,10 @@ runs:
             --helm-set=hubble.relay.image.useDigest=false \
             --helm-set=hubble.eventBufferCapacity=65535 \
             --helm-set=bpf.monitorAggregation=none \
-            --helm-set=authentication.mutual.spire.enabled=true \
+            --helm-set=cluster.name=default \
+            --helm-set=authentication.mutual.spire.enabled=${{ inputs.mutual-auth }} \
+            --helm-set=bpfClockProbe=${{ inputs.bpf-clock-probe }} \
+            --helm-set=cni.uninstall=false \
```
Comment (aanm) [inline .github/actions/cilium-config/action.yml]: Wait, why is this false? I read the PR description but if our intention is to replicate user-installations we should change defaults to accommodate our tests.

**Review by aanm — CHANGES_REQUESTED** [review]

File: .github/actions/cilium-config/action.yml (hunk 1 above)
Comment (brb) [reply .github/actions/cilium-config/action.yml]: My reading of the flag is that in most test cases it's a noop. But yeah, let's make it configurable.

**Review by aanm — APPROVED** [review]

**Comment by brb** [comment]
ci-e2e passed, ready to merge!

**Pull request merged**